docs/WalletGuide.md

```diff
@@ -0,0 +1,301 @@
+
+Authenticate an account with the anchor using SEP-10.
+
+```typescript
+// create an account to authenticate
+let account = wal.stellar().account();
+let accountKp = account.createKeypair();
+
+// authenticate with Sep-10
+let auth = await anchor.auth();
+let authToken = await auth.authenticate({ accountKp });
+```
+
+Get available anchor services and information about them. For example, interactive deposit/withdrawal limits, currency, fees, payment methods.
+
+```typescript
+let serviceInfo = await anchor.getServicesInfo();
+```
+
+Interactive deposit and withdrawal.
+
+```typescript
+const assetCode = "XLM";
+
+let resp = await anchor.interactive().deposit({
+  accountAddress: accountKp.publicKey(),
+  assetCode,
+  authToken,
+});
+
+let resp = await anchor.interactive().withdraw({
+  accountAddress: accountKp.publicKey(),
+  assetCode,
+  authToken,
+});
+```
+
+Get a single transaction's current status and details.
+
+```typescript
+const transaction = await anchor.getTransactionBy({
+  authToken,
+  id: transactionId,
+});
+```
+
+Get account transactions for a specified asset.
+
+```typescript
+const transactions = await anchor.getHistory({
+  authToken,
+  assetCode,
+});
+```
+
+Watch transaction.
+
+```typescript
+let watcher = anchor.watcher();
+
+let { stop } = watcher.watchOneTransaction({
+  authToken,
+  assetCode,
+  id: successfulTransaction.id,
+  onMessage,
+  onSuccess,
+  onError,
+});
+
+// stop watching transaction
+stop();
+```
+
+### Add client domain signing
+
+One of the features being supported
+by [SEP-10 Authentication](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md)
+is verifying client domain. This enables anchor to recognize that user request was made using a specific client.
+
+Functionality of signing with client domain is enabled by the [WalletSigner](../wallet-sdk/src/walletsdk/Auth/WalletSigner.ts). You can use the `DefaultSigner` class as a baseline for your implementation. The main method that needs to be implemented is `signWithDomainAccount`.
+
+Implementation should make a request to a backend server containing the client domain private key. Server returns signed transaction, finishing the flow.
+
+For your wallet signer to be used, it's required to set it in this SDK. It's recommended to set it globally via `ApplicationConfiguration` `defaultSigner` [option](../wallet-sdk/src/walletsdk/index.ts).
+Alternatively, it can be passed to the `authenticate` call of the `Auth` class.
+
+#### Example
+
+In the example below, [Stellar demo wallet](https://demo-wallet.stellar.org) will be used as a client domain. Server-side implementation, responsible for signing transaction can be found [here](https://github.com/stellar/stellar-demo-wallet/tree/master/packages/demo-wallet-server).  
+
+First, create an object using the `WalletSigner` interface, and define both the signWithClientAccount and a signWithDomainAccount method.
+
+```typescript
+const demoWalletSigner: WalletSigner = {
```

Comment:
There should be a default implementation people can extend from
Also, there should be a default implementation that already has `axios.post` implementation

Reply:
> There should be a default implementation people can extend from
for the ts-sdk they can just create their own instance that inherits from `WalletSigner` interface, instead of needing to create their own new class. Unless I'm missing something?

>Also, there should be a default implementation that already has axios.post implementation
yup we have that [open ticket](https://stellarorg.atlassian.net/browse/WAL-882?atlOrigin=eyJpIjoiZjcwZWNhNWI5NGE4NGEwNmI2NTljOTc3ZGViNzY5ZWIiLCJwIjoiaiJ9) for that still, but I can work on that next if it's needed ASAP and then update these docs